src/owner.rs

```diff
@@ -0,0 +1,357 @@
+use {Model, User};
+use util::{RequestUtils, CargoResult, internal, ChainError, human};
+use db::Connection;
+use curl::http;
+use pg::rows::Row;
+use rustc_serialize::json;
+use util::errors::NotFound;
+use std::str;
+
+
+/// Unifies the notion of a User or a Team.
+pub enum Owner {
+    User(User),
+    Team(Team)
+}
+
+/// For now, just a Github Team. Can be upgraded to other teams
+/// later if desirable.
+pub struct Team {
+    /// Github annoyingly has some APIs talk about teams by name,
+    /// and others by some opaque id. I couldn't find any docs
+    /// suggesting these ids are stable, so let's just always
+    /// ask github for it. *shrug*
+    github_id: i32,
+    /// Unique table id
+    cargo_id: i32,
+    /// "github:org:team"
+    /// An opaque unique ID, that was at one point parsed out to query Github.
+    /// We only query membership with github using the github_id, though.
+    name: String,
+}
+
+#[derive(RustcDecodable, RustcEncodable)]
+pub struct EncodableOwner {
+    pub id: i32,
+    // Login is deprecated in favour of name, but needs to be printed for back-compat
+    pub login: String,
+    pub kind: String,
+    pub email: Option<String>,
+    pub name: Option<String>,
+    pub avatar: Option<String>,
+}
+
+/// Access rights to the crate (publishing and ownership management)
+/// NOTE: The order of these variants matters!
+#[derive(PartialEq, Eq, PartialOrd, Ord)]
+pub enum Rights {
+    None,
+    Publish,
+    Full,
+}
+
+impl Team {
+    /// Just gets the Team from the database by name.
+    pub fn find_by_name(conn: &Connection, name: &str) -> CargoResult<Self> {
+        let stmt = try!(conn.prepare("SELECT * FROM teams
+                                      WHERE name = $1"));
+        let rows = try!(stmt.query(&[&name]));
+        let row = try!(rows.iter().next().chain_error(|| {
+            NotFound
+        }));
+        Ok(Model::from_row(&row))
+    }
+
+    /// Tries to create the Team in the DB (assumes a `:` has already been found).
+    pub fn create(conn: &Connection, name: &str, req_user: &User) -> CargoResult<Self> {
+        // must look like system:xxxxxxx
+        let mut chunks = name.split(":");
+        match chunks.next().unwrap() {
+            // github:rust-lang:owners
+            "github" => {
+                // Ok to unwrap since we know one ":" is contained
```

Comment:
This is actually going to revert back -- the try! on org is otherwise dead code.